### PR TITLE
Add recursive deletion and UI updates for root directory management

### PR DESCRIPTION
**Pull request by shinshin86**
Hello!

Thank you for providing such a useful tool; I use it regularly and find it incredibly convenient. While using it, I thought of a feature that could make it even more useful, so I’ve implemented and added it in this PR.

## Summary of Changes

This pull request introduces the following changes: 
1. **Recursive Deletion of Root Directory Contents:** 
- Added the `deleteDirectoryContentsRecursive` function in `contentscript.js` to recursively delete all files and subdirectories within a directory. 
- Implemented the `deleteRoot` message handler to allow users to delete all contents in the root directory. 

2. **UI Updates for Root Directory Actions:** 
- Added a "Delete All" button to the root directory in the tree structure displayed in the DevTools panel (`devtools.js`). 
add-recursive-deletion_1

- Adjusted button visibility and functionality:
- Buttons are hidden when the root directory is empty. 
add-recursive-deletion_3

- The "Delete All" button prompts the user for confirmation before executing the deletion.
add-recursive-deletion_2

- After deletion, the tree structure refreshes to reflect the updated state. 

3. **Code Cleanup:** 
- Removed redundant elements and adjusted DOM manipulations for better consistency, such as replacing `div.remove()` with `details.remove()`.

## Questions for Reviewers

As an external contributor, I aim to align my contributions with the project's vision and maintainers' preferences. The key question for this PR is whether the proposed UI changes, particularly the addition of the "Delete All" button for the root directory, align with the intended direction of this tool. 

Specifically: 
- **User Experience:** Are the new buttons and their behaviors (visibility toggling, confirmation dialogs) acceptable from a usability perspective? 
- **Scope of Functionality:** Does the ability to delete all contents in the root directory fit within the tool's intended use cases? 

## Notes

I am happy to adjust the implementation or the UI/UX based on your feedback and suggestions. Thank you for taking the time to review this PR!

### CODE REVIEW

File: devtools.js [hunk 1]
```diff
@@ -42,6 +42,73 @@
           details.open = true;
           details.classList.add('root');
           summary.textContent = ' ';
+
+          const rootNameSpan = document.createElement('span');
+          rootNameSpan.classList.add('directory-name');
+          rootNameSpan.textContent = 'Root';
```
Comment (tomayac) [inline devtools.js]: This now creates duplication: `(Root) Root`:

image

File: devtools.js [hunk 2]
```diff
@@ -42,6 +42,73 @@
...
+                    errorDialog.querySelector('p').textContent = response.error;
+                    return errorDialog.showModal();
+                  }
+                },
+              );
+            });
+
+            deleteButton.addEventListener('click', (event) => {
+              confirmDialog.querySelector('span').textContent = 'all files and directories';
+              confirmDialog.querySelector('code').textContent = 'Root';
```
Comment (tomayac) [inline devtools.js]: Careful, the English here isn't correct: "Do you really want to delete <del>the</del> all files and directories <ins>in</ins> `Root`?" (see the insertion and deletion):
 
image

**Review by tomayac — CHANGES_REQUESTED** [review]
Thanks so much for further improving this extension. I have left two comments that should be addressed, but after that, I'm more than happy to merge this. Thanks again!

File: devtools.js (hunk 2 above)
Comment (shinshin86) [reply devtools.js]: @tomayac 
Thank you for the review! I’ve updated the wording to:
"Do you really want to delete the Root directory?"
Does this look better to you?
opfs-explorer_pull_22

**Comment by shinshin86** [comment]
@tomayac Thank you so much for reviewing!
I’ve pushed the changes, and I’ve also replied to one of the comments, so I’d appreciate it if you could check that as well.

If you have any questions or concerns about my changes, please feel free to ask.

Thank you!

File: devtools.js [hunk 3]
```diff
@@ -41,7 +41,69 @@
         if (value.relativePath === '.') {
           details.open = true;
           details.classList.add('root');
-          summary.textContent = ' ';
+          summary.textContent = ' (Root)';
```
Comment (tomayac) [inline devtools.js]:
```suggestion
          summary.textContent = '(Root)';
```

File: devtools.js [hunk 4]
```diff
@@ -41,7 +41,69 @@
...
+                  if (response.error) {
+                    errorDialog.querySelector('p').textContent = response.error;
+                    return errorDialog.showModal();
+                  }
+                },
+              );
+            });
+
+            deleteButton.addEventListener('click', (event) => {
+              confirmDialog.querySelector('code').textContent = 'Root directory';
```
Comment (tomayac) [inline devtools.js]:
```suggestion
              confirmDialog.querySelector('code').textContent = '🏠 (Root)';
```

**Review by tomayac — APPROVED** [review]
LGTM, thank you!

Made some minor changes, so it's `🏠 (Root)` everywhere.

**Comment by shinshin86** [comment]
Got it! Thank you so much!

**Comment by tomayac** [comment]
This should now be released as `v1.8.0` on the Chrome Web Store. Thank _you_ again!

**Pull request merged**